sklearn/covariance/_shrunk_covariance.py

```diff
@@ -19,10 +19,89 @@
 
 from . import empirical_covariance, EmpiricalCovariance
 from .._config import config_context
-from ..utils import check_array
+from ..utils import check_array, as_float_array
 from ..utils._param_validation import Interval
 
 
+def _ledoit_wolf_shrinkage(X, assume_centered, block_size):
```

Comment:
I would let this function as is.

I assume that we wanted it private. However, we would need a deprecation it if we do this change.

Reply:
But if it stays as is, then we will be checking
https://github.com/scikit-learn/scikit-learn/blob/64432e1cf65fc87c763737aa6422bb92d3573786/sklearn/covariance/_shrunk_covariance.py#L236-L239
twice - once in the class and in the shrinkage estimation

Reply:
No you just need to remove it from the shrinkage estimation because we already did it.